test/job/adapters/slurm_test.rb

```diff
@@ -0,0 +1,24 @@
+require 'test_helper'
+
+class TestSlurm < Minitest::Test
+  include TestHelper
+
+  def slurm_instance(config = {})
+    OodCore::Job::Factory.build({ adapter: 'slurm' }.merge(config))
+  end
+
+  def test_submit_interface
+    slurm = slurm_instance
+
+    assert(slurm.respond_to?(:submit))
+    veryify_keywords(slurm, :submit, [:after, :afterok, :afternotok, :afterany])
+    verify_args(slurm, :submit, 1)
+  end
+
+  def test_submitting_with_hold
+    slurm = slurm_instance
+    stub_submit
+    OodCore::Job::Adapters::Slurm::Batch.any_instance.expects(:submit_string).with(script_content, args: ["-H", "--export", "NONE"], env: {})
+    slurm.submit(build_script(submit_as_hold: true))
+  end
```

Comment:
This test replaces this rspec test below, and just reinforces my issue with rspec.  Here in mintest we see what I have to stub Open3 because at some point calling `submit` on an adapter will result in an `Open3.capture3` call. 

I guess I just like the explicit nature of Minitest. How is rspec stubbing Open3? I guess it doesn't need to? Like somehow calling `submit` will stop execution once `slurm.submit_string` is called? There's just a bit too much magic happening, to the point where I'm not really even sure what code is being executed during this test.

https://github.com/OSC/ood_core/blob/65f8ac5dcebf6b4546d1082c308b724c26e2eca3/spec/job/adapters/slurm_spec.rb#L72-L77